libglusterfs/src/logging.c

```diff
@@ -443,7 +444,10 @@ gf_openlog(const char *ident, int option, int facility)
     }
 
     /* TODO: Should check for errors here and return appropriately */
-    setlocale(LC_ALL, "");
```

Comment:
On looking at the man page, it appears that  `setlocale(LC_ALL, "") ` is intentional and correct.  The way I understood it, a portable program sets it this way so that the user can choose the required localization by setting the relevant environment variables and the error messages will appear in that locale. If we hard-code it to "en_US.UTF-8", then those variables won't take effect.

Reply:
This is the whole point. I cannot look at logs with Japanese errors. I assume all developers, who look at logs, know English.

Reply:
What if the cluster admin wants the errors in Japanese though?

Reply:
The logs (as I see it) are not for the admin, but for developers to understand what's going on.
In the balance between helping users who have issues and the admin who can't read English, I favor the former.

Reply:
Just looked at some other projects, all of them seem to be preferring to give control to the end user where it makes sense:

```
[ravi@linuxpad ceph]$ git grep setlocale
src/common/compat.cc:  input.imbue(std::locale(setlocale(LC_ALL, nullptr)));

[ravi@linuxpad qemu]$ git grep setlocale
configure:  setlocale(LC_ALL, "");
tests/test-qga.c:    setlocale (LC_ALL, "");
ui/curses.c:    setlocale(LC_CTYPE, "");
ui/gtk.c:    setlocale(LC_MESSAGES, "");
ui/gtk.c:    setlocale(LC_CTYPE, "C.UTF-8");
ui/gtk.c:     * Since GTK+ calls setlocale() by default -importing the locale
ui/gtk.c:     * using gtk_disable_setlocale().
ui/gtk.c:     * LC_MESSAGES from the environment (see the setlocale() call
ui/gtk.c:    gtk_disable_setlocale();

[ravi@linuxpad samba]$ git grep setlocale
lib/util/charset/codepoints.c:  setlocale(LC_ALL, "");
source3/rpc_server/samr/srv_samr_chgpasswd.c:           const char *prevlocale = setlocale(LC_ALL, "C");
source3/rpc_server/samr/srv_samr_chgpasswd.c:           setlocale(LC_ALL, prevlocale);
source3/utils/net.c:    setlocale(LC_ALL, "");
source3/wscript:    conf.CHECK_FUNCS('syslog vsyslog timegm setlocale')
source3/wscript:setlocale
source4/dsdb/tests/python/sort.py:locale.setlocale(locale.LC_ALL, ('en_US', 'UTF-8'))
source4/heimdal/kuser/kinit.c:    setlocale (LC_ALL, "");
```
IMHO, we could have this fix downstream for RHGS but I believe for upstream where admins (not developers) do turn to logs if CLI doesn't give any clue, it is good to give a choice.

Reply:
Not sure how @xhernandez @amarts  feel.  I'm ok to remove -1 if everyone feels its good to have it in English only :+1: 
Apologies for the late feedback (the issue seems to have been raised long back).

Reply:
The idea of bringing the message ID in the logging framework was to support locale (and that is where it got introduced in code). So, the code helps to understand the error for developers, and sys-admins!

But considering the effort is not complete across codebase, this may be OK change (ie, only English). IMO, we should be having tools to identify logs and issues based on message Id. If we have such tools, its good to have 'locale' support... because the project has more chances for adoption if we support locales.